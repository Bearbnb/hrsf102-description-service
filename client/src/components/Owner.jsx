import React from 'react';
import PropTypes from 'prop-types';

const Owner = ({ ownerName, ownerPicture }) => (
  <div>
    <div>{ownerName}</div>
    <img src={ownerPicture} alt="" />
  </div>
);

Owner.propTypes = {
  ownerName: PropTypes.string,
  ownerPicture: PropTypes.string,
};

Owner.defaultProps = {
  ownerName: 'John Smith',
  ownerPicture: 'https://s3-us-west-1.amazonaws.com/airbnb-owner-photos/airbnb1.jpg',
};

export default Owner;
