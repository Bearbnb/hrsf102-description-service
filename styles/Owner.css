.container {
  max-width: 64px;
  grid-column-start: 2;
  grid-column-end: 3;
  grid-row-start: 2;
  grid-row-end: 3;
}

.imgCircle {
  border-radius: 50%;
  height: 64px;
  width: 64px;
  margin-left: auto;
  margin-right: auto;
}

.name {
  margin: 0 auto 0 auto;
  text-align: center;
  font-family: Circular,"Helvetica Neue",Helvetica,Arial,sans-serif;
  font-size: 14px;
  color: #808080;
}