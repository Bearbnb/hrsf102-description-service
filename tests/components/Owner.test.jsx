import { shallow } from 'enzyme';
import React from 'react';
import toJson from 'enzyme-to-json';
import Owner from '../../client/src/components/Owner';

describe('<Owner />', () => {
  it('should render properly', () => {
    const wrapper = shallow(<Owner amenities={{}} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
