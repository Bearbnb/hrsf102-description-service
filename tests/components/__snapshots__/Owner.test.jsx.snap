// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Owner /> should render properly 1`] = `
<div
  className="container"
>
  <img
    alt=""
    className="imgCircle"
    src="https://s3-us-west-1.amazonaws.com/airbnb-owner-photos/airbnb1.jpg"
  />
  <div
    className="name"
  >
    John
  </div>
</div>
`;
